akka-http-core/src/main/java/akka/http/javadsl/model/Multipart.java

```diff
@@ -0,0 +1,145 @@
+/**
+ * Copyright (C) 2009-2014 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+package akka.http.javadsl.model;
+
+import java.util.Map;
+import scala.concurrent.Future;
+import akka.http.javadsl.model.headers.ContentDisposition;
+import akka.http.javadsl.model.headers.ContentDispositionType;
+import akka.http.javadsl.model.headers.RangeUnit;
+import akka.japi.Option;
+import akka.stream.Materializer;
+import akka.stream.javadsl.Source;
+
+/**
+ * The model of multipart content for media-types `multipart/\*` (general multipart content),
+ * `multipart/form-data` and `multipart/byteranges`.
+ *
+ * The basic modelling classes for these media-types ([[Multipart.General]], [[Multipart.FormData]] and
+ * [[Multipart.ByteRanges]], respectively) are stream-based but each have a strict counterpart
+ * (namely [[Multipart.General.Strict]], [[Multipart.FormData.Strict]] and [[Multipart.ByteRanges.Strict]]).
+ */
+public interface Multipart {
+
+    MediaType.Multipart getMediaType();
+
+    Source<? extends Multipart.BodyPart, Object> getParts();
+
+    /**
+     * Converts this content into its strict counterpart.
+     * The given `timeout` denotes the max time that an individual part must be read in.
+     * The Future is failed with an TimeoutException if one part isn't read completely after the given timeout.
+     */
+    Future<? extends Multipart.Strict> toStrict(long timeoutMillis, Materializer materializer);
+
+    /**
+     * Creates an entity from this multipart object.
+     */
+    RequestEntity toEntity(HttpCharset charset, String boundary);
+
+    interface BodyPart {
+        BodyPartEntity getEntity();
+
+        Iterable<HttpHeader> getHeaders();
+
+        Option<ContentDisposition> getContentDispositionHeader();
+
+        Map<String, String> getDispositionParams();
+
+        Option<ContentDispositionType> getDispositionType();
+
+        Future<? extends Multipart.BodyPart.Strict> toStrict(long timeoutMillis, Materializer materializer);
+
+        interface Strict extends Multipart.BodyPart {
+            HttpEntity.Strict getEntity();
+        }
+    }
+
+    interface Strict extends Multipart {
```

Comment:
I’d move this before `BodyPart`: it took me some seconds to see that this is a MultiPart and not a MultiPart.BodyPart subtype.